src/main/java/com/bobocode/hoverla/bring/context/BeanValidator.java

```diff
@@ -1,18 +1,113 @@
 package com.bobocode.hoverla.bring.context;
 
-import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import com.bobocode.hoverla.bring.exception.BeanValidationException;
+import lombok.extern.slf4j.Slf4j;
 
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 /**
- * Validates beans list to avoid duplicates or other conflicts
+ * Validates beans list to avoid such cases: duplicate names, bean name is null, empty, or contains spaces, bean type
+ * is not suitable for the required type, circular dependency, application context doesn't contain the required bean
```

Comment:
(nit) what if to list each case from new line and start them with some symbol (like * or -) ?